### PR TITLE
Add internal support for silent authentication

### PR DESCRIPTION
**Pull request by NSeydoux**
This enables to specify the "prompt" option from our own internal API
all the way down to the OIDC library we are using. The "prompt" option,
when set to 'none', prevents the Identity Provider to display any screen
or to have any sort of user interaction, which is what we want for
silent authentication.

Note that the "prompt" option was not added to
the external API, because there is no reason for the calling library to
decide it needs to go through silent authentication: we make this call
entierly based on internal state. The code to implement this logic of
whether we want or not to go through silent authentication is not part
of this PR.

# Checklist

- [X] All acceptance criteria are met.
- [X] Relevant documentation, if any, has been written/updated.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: packages/browser/__tests__/login/oidc/oidcHandlers/AuthorizationCodeWithPkceOidcHandler.spec.ts [hunk 1]
```diff
@@ -129,7 +133,27 @@ describe("AuthorizationCodeWithPkceOidcHandler", () => {
       );
     });
 
+    it("supports silent authentication", async () => {
```
Comment (pmcb55) [inline packages/browser/__tests__/login/oidc/oidcHandlers/AuthorizationCodeWithPkceOidcHandler.spec.ts:136]: Seems this test is more specifically about setting an option and asserting that the low-level library is called with it
```suggestion
    it("passes our the 'prompt' option down to our OIDC client library implementation", async () => {
```

File: packages/browser/src/ClientAuthentication.ts [hunk 2]
```diff
@@ -70,7 +70,12 @@ export default class ClientAuthentication {
   // Isn't Javascript fun?
   login = async (
     sessionId: string,
-    options: ILoginInputOptions
+    options: ILoginInputOptions & {
+      // TODO: the 'prompt' parameter shouldn't be exposed as part of the public API.
+      // This classe's login should take the internal IOidcOptions type as an
```
Comment (pmcb55) [inline packages/browser/src/ClientAuthentication.ts:75]:
```suggestion
      // This login function should take the internal IOidcOptions type as an
```

File: packages/core/src/login/ILoginOptions.ts [hunk 3]
```diff
@@ -38,7 +38,10 @@ export default interface ILoginOptions extends ILoginInputOptions {
   // 'bundling' into this options interface), but it wouldn't be a significant
   // improvement really...
   sessionId: string;
-
+  /**
+   * Specify wether the Solid Identity Provider may or may interact with the user.
```
Comment (pmcb55) [inline packages/core/src/login/ILoginOptions.ts:42]:
```suggestion
   * Specify whether the Solid Identity Provider may, or may not, interact with the user (for example,
   * the normal login process **_requires_** human interaction for them to enter their credentials,
   * but if a user simply refreshes the current page in their browser, we'll want to log them in again
   * automatically, i.e., without prompting them to manually provide their credentials again).
```

File: packages/core/src/login/oidc/IOidcOptions.ts [hunk 4]
```diff
@@ -33,28 +33,31 @@ import { IClient } from "./IClient";
...
   sessionId: string;
   refreshToken?: string;
-}
-
-/**
- * @hidden
- */
-export interface IAccessTokenOidcOptions extends ICoreOidcOptions {
+  /**
+   * Specify wether the Solid Identity Provider may or may interact with the user.
```
Comment (pmcb55) [inline packages/core/src/login/oidc/IOidcOptions.ts:49]: Ditto to the comment above - same typos here too...

File: packages/core/src/login/oidc/IOidcOptions.ts [hunk 5]
```diff
@@ -33,28 +33,31 @@ import { IClient } from "./IClient";
...
+   */
+  prompt?: string;
+  /**
+   * True if a dpop compatible auth_token should be requested.
```
Comment (pmcb55) [inline packages/core/src/login/oidc/IOidcOptions.ts:53]:
```suggestion
   * True if a DPoP compatible auth_token should be requested.
```

File: packages/core/src/login/oidc/IOidcOptions.ts [hunk 6]
```diff
@@ -33,28 +33,31 @@ import { IClient } from "./IClient";
...
+   */
   dpop: boolean;
+  /**
+   * The URL to which the user should be redirected after authorizing.
```
Comment (pmcb55) [inline packages/core/src/login/oidc/IOidcOptions.ts:57]:
```suggestion
   * The URL to which the user should be redirected after logging in and 
   * authorizing our app to access their Pod.
```

**Review by pmcb55 — APPROVED** [review]

File: packages/core/src/login/oidc/IOidcOptions.ts (hunk 4 above)
Comment (NSeydoux) [reply packages/core/src/login/oidc/IOidcOptions.ts]: 2f7f4b921cbb20661cc1adccba736312f7e587da

File: packages/core/src/login/oidc/IOidcOptions.ts (hunk 6 above)
Comment (NSeydoux) [reply packages/core/src/login/oidc/IOidcOptions.ts]: I rephrased this slightly in 2f7f4b921cbb20661cc1adccba736312f7e587da, just to be extra clear that logging in refers in this context to the user logging in the IdP

**Pull request merged**